MH4U Dex/CoreDataController.m

```diff
@@ -0,0 +1,446 @@
+
+#pragma mark - Area Drop Constants
+
+NSString *const AreaDropAreaNameKey = @"area";
+NSString *const AreaDropRankKey = @"rank";
+NSString *const AreaDropIDKey = @"_id";
+NSString *const AreaDropMethodKey = @"method";
+NSString *const AreaDropPercentKey = @"chance";
+NSString *const AreaDropQuantityKey = @"number";
+NSString *const AreaDropItemNameKey = @"item";
+NSString *const AreaDropIDDecimalStringKey = @"idDecimalString";
+
+#pragma mark - Singleton Method
+
++ (CoreDataController *)sharedCDController
+{
+    static CoreDataController *sharedInstance = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        sharedInstance = [[CoreDataController alloc] init];
+    });
+    return sharedInstance;
+}
+
+#pragma mark - Initial Data Loading
+
+- (void)loadMonsterData
+{
+    // Because monsters are added first, there are no relationships to set up, so we can hydrate them.
+    NSString *path = [[NSBundle mainBundle] pathForResource:MonstersFileName ofType:JSON];
+    NSDictionary *attributes = @{
+                                 @"id":@"_id",
+                                 @"monster_class":@"class",
+                                 @"name":@"name",
+                                 @"jpn_name":@"jpn_name",
+                                 @"sort_name":@"sort_name",
+                                 @"trait":@"trait",
+                                 @"icon":@"icon_name",
+                                 @"url":@"url"
+                                 };
+    [self.managedObjectContext hydrateStoreWithJSONAtPath:path attributeMappings:attributes forEntityName:[Monster entityName]];
+    path = [[NSBundle mainBundle] pathForResource:MonsterDamageZonesFileName ofType:JSON];
+    attributes = @{
+                   @"id":@"_id",
+                   @"monster_id":@"monster_id",
+                   @"monsterName":@"monster_name",
+                   @"jpnMonsterName":@"jpn_monster_name",
+                   @"bodyPart":@"body_part",
+                   @"cut":@"cut",
+                   @"impact":@"impact",
+                   @"shot":@"shot",
+                   @"fire":@"fire",
+                   @"water":@"water",
+                   @"thunder":@"thunder",
+                   @"ice":@"ice",
+                   @"dragon":@"dragon",
+                   @"ko":@"ko",
+                   @"extract":@"extract"
+                   };
+    [self.managedObjectContext hydrateStoreWithJSONAtPath:path attributeMappings:attributes forEntityName:[DamageZone entityName]];
+}
+
+- (void)loadMonsterDropData
+{
+    NSArray *dropList = [self loadArrayFromJsonFileNamed:MonsterDropsFileName];
+    for (NSDictionary *drop in dropList) {
+        // Only add the drop if it is not already there
+        NSNumber *monsterDropID = drop[MonsterDropIDKey];
+        NSPredicate *monsterDropPredicate = [NSPredicate predicateWithFormat:@"%K == %d", MonsterDropIDKey, monsterDropID.intValue];
```

Comment:
For things that are properties of your core data model entity, you should have a constant provided by mogenerator—in this case, something like `MonsterAttributes.id`.